File: engine/model/src/main/scala/tools/packet/statusinfo/ShowEXPGain.scala
package tools.packet.statusinfo

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class ShowEXPGain(private var _gain: Int, private var _equip: Int, private var _party: Int,
                  private var _inChat: Boolean, private var _white: Boolean) extends PacketInput {
  def gain: Int = _gain

  def equip: Int = _equip

  def party: Int = _party

  def inChat: Boolean = _inChat

  def white: Boolean = _white

  override def opcode(): SendOpcode = SendOpcode.SHOW_STATUS_INFO
}

File: engine/engine-model/src/main/java/tools/packet/statusinfo/ShowEXPGain.java
package tools.packet.statusinfo;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record ShowEXPGain(Integer gain, Integer equip, Integer party, Boolean inChat,
                          Boolean white) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.SHOW_STATUS_INFO;
   }
}